lib/astutils.cpp

```diff
@@ -133,9 +133,74 @@ const Token * astIsVariableComparison(const Token *tok, const std::string &comp,
     return ret;
 }
 
+const Token * getVariableExpression(const Variable * var)
+{
+    if(!var || !var->declEndToken())
+        return nullptr;
+    if(Token::Match(var->declEndToken(), "; %varid% =", var->declarationId()))
+        return var->declEndToken()->tokAt(2)->astOperand2();
+    else
+        return var->declEndToken()->astOperand2();
+}
+
+bool isInLoop(const Token * tok) {
+    while(tok->astParent()) {
+        tok = tok->astParent();
+        if(Token::Match(tok->astOperand1(), "for|while"))
+            return true;
+    }
+    return false;
+}
+
+const Token * followVariableExpression(const Token * tok, bool cpp)
```

Comment:
> The logic feels somewhat ValueFlow related but for now I think you can keep it here.

Yea, except its tracking expressions instead of values. In the future, we may want to have a pass that sets these results in the `Token` class.

Reply:
> In the future, we may want to have a pass that sets these results in the Token class.

Yes.